packages/cli/src/scaling/worker-server.ts

```diff
@@ -0,0 +1,137 @@
+import { GlobalConfig } from '@n8n/config';
+import express from 'express';
+import { ensureError } from 'n8n-workflow';
+import http from 'node:http';
+import type { Server } from 'node:http';
+import { Service } from 'typedi';
+
+import config from '@/config';
+import { CredentialsOverwrites } from '@/credentials-overwrites';
+import * as Db from '@/db';
+import { CredentialsOverwritesSetError } from '@/errors/credentials-overwrites-set.error';
+import { NonJsonBodyError } from '@/errors/non-json-body.error';
+import { NonWorkerInstanceTypeError } from '@/errors/non-worker-instance-type.error';
+import { PortTakenError } from '@/errors/port-taken.error';
+import { ServiceUnavailableError } from '@/errors/response-errors/service-unavailable.error';
+import { ExternalHooks } from '@/external-hooks';
+import type { ICredentialsOverwrite } from '@/interfaces';
+import { Logger } from '@/logger';
+import { rawBodyReader, bodyParser } from '@/middlewares';
+import * as ResponseHelper from '@/response-helper';
+import { ScalingService } from '@/scaling/scaling.service';
+
+/**
+ * Responsible for handling HTTP requests sent to a worker.
+ */
+@Service()
+export class WorkerServer {
+	private port: number;
+
+	private server: Server;
+
+	/**
+	 * @doc https://docs.n8n.io/embed/configuration/#credential-overwrites
+	 */
+	private overwritesLoaded = false;
+
+	constructor(
+		private readonly globalConfig: GlobalConfig,
+		private readonly logger: Logger,
+		private readonly scalingService: ScalingService,
+		private readonly credentialsOverwrites: CredentialsOverwrites,
+		private readonly externalHooks: ExternalHooks,
+	) {
+		this.assertWorker();
+
+		const app = express();
+
+		app.disable('x-powered-by');
+
+		this.server = http.createServer(app);
+
+		this.port = this.globalConfig.queue.health.port;
+
+		const overwritesEndpoint = this.globalConfig.credentials.overwrite.endpoint;
+
+		this.server.on('error', (error: NodeJS.ErrnoException) => {
+			if (error.code === 'EADDRINUSE') throw new PortTakenError(this.port);
+		});
+
+		if (this.globalConfig.queue.health.active) {
+			app.get('/healthz', async (req, res) => await this.healthcheck(req, res));
+		}
+
+		if (overwritesEndpoint !== '') {
+			app.post(`/${overwritesEndpoint}`, rawBodyReader, bodyParser, (req, res) =>
+				this.handleOverwrites(req, res),
+			);
+		}
+	}
+
+	async init() {
+		await new Promise<void>((resolve) => this.server.listen(this.port, () => resolve()));
```

Comment:
Unnecessary function wrapping:
```suggestion
		await new Promise<void>((resolve) => this.server.listen(this.port, resolve));
```
Or is there a type error if done this way?